flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/OptionSerializer.scala

```diff
@@ -65,7 +65,7 @@ class OptionSerializer[A](val elemSerializer: TypeSerializer[A])
     case Some(a) =>
       target.writeBoolean(true)
       elemSerializer.serialize(a, target)
-    case None =>
+    case _ =>
```

Comment:
If `writeBoolean` is correctly implemented (i.e., according to the JavaDocs in `java.io.DataOutput#writeBoolean(boolean)`) we should be good. However, you are right, that we may not change the `0` -> `false` and `1` -> `true` mapping.